src/sonic-eventd/src/eventd.cpp

```diff
@@ -343,8 +346,10 @@ capture_service::init_capture_cache(const event_serialized_lst_t &lst)
             sequence_t seq;
 
             if (validate_event(event, rid, seq)) {
-                m_pre_exist_id[rid] = seq;
-                m_events.push_back(*itc);
+                if (itc->find("heartbeat") == string::npos && VEC_SIZE(m_events) < m_cache_max) { // filter out heartbeat events and bound m_events
+                    m_pre_exist_id[rid] = seq;
```

Comment:
>m_pre_exist_id [](http://example.com/codeflow?start=20&length=14)

Since m_pre_exist_id is also possible increasing size, would you set uplimit, and check size before this line?

Reply:
Done